docs/proposal-universal-id.md

```diff
@@ -0,0 +1,324 @@
+"objects" to operate on:
+
+* `[CYCLE-POINT-GLOB/]TASK-NAME-GLOB[:TASK-STATE]`
+* `[CYCLE-POINT-GLOB/]FAMILY-NAME-GLOB[:TASK-STATE]`
+* `TASK-NAME-GLOB[.CYCLE-POINT-GLOB][:TASK-STATE]`
+* `FAMILY-NAME-GLOB[.CYCLE-POINT-GLOB][:TASK-STATE]`
+
+With the GraphQL schema came the requirement for a global identifier
+which added yet another format:
+
+* `[user[|]]workflow[|cycle[|(namespace)[|job]]]`
+
+## Motivating Factors For This Proposal
+
+1) The global identifier is nicer and more flexible than the command line
+   options.
+2) It is not possible to specify jobs with any of the CLI formats.
+   https://github.com/cylc/cylc-flow/pull/3880
+3) The [rose suite-run migration proposal](https://github.com/cylc/cylc-admin/blob/master/docs/proposal-rose-suite-run.md#cli-changes)
+   suggests converting the `REG` (read `FLOW`) arguments into options i.e:
+
+   ```console
+   $ cylc stop --flow my_workflow
+   ```
+
+   In order to make the edge where users wish to operate on a workflow and
+   are already, conveniently, in the run directory of that workflow slightly
+   nicer:
+
+   ```console
+   $ cd ~/cylc-run/my_workflow
+   $ cylc stop
+   ```
+4) The UI currently doesn't make use of any form of identifier, however,
+   it may be beneficial to expose users to the global identifier e.g.
+   for mutations
+5) The CLI is now implemented via GraphQL anyway meaning that the CLI arguments
+   are provided as GraphQL arguments and transformed as required.
+
+## Proposed Universal Identifier
+
+resolves https://github.com/cylc/cylc-flow/issues/3592
+
+* Use `/` as the separator (as originally intended) but use `//` to separate
+  the flow from the cycle.
+* Permit globs in each part of the id.
+* Support `:<arg>` as a special state selector for convenience in each section
+  other than "user".
+  (i.e. a direct replacement for the current `cylc trigger flow *:failed`
+  syntax).
+
+```bash
+# regular form (for identifying objects)
+[~user[/]]flow//cycle/task/job
+
+# expanded form with selectors (for quick querying of objects)
+[~user[/]]flow[:state]//cycle[:state]/task[:state]/job[:state]
+```
+
+Examples (equivalent GraphQL arguments under current scheme)
+
+```bash
+# flow under account user (wid="user|flow")
+~user/flow
+
+# the task foo at all cycles in flow (wid="flow|*|foo")
+flow//*/foo
+
+# all tasks in cycles between (exclusive) 202009 202011 in flow
+# (wid="flow|202010*|*)
+flow//202010*/*
+
+# all tasks begining foo in flow (wid="flow|*|foo*)
+flow//*/foo*
+
+# all running workflows (wid="*", state="running")
+*:running
+
+# all cycles in foo with at least one running task
+# (wid="flow|*", state="running")
+flow//*:running
+
+# all running tasks in foo (wid="flow|*|*", state="running")
+flow//*/*:running
+```
+
+> **Note:** All syntax implemented here represents functionality currently
+  supported in the GraphQL schema.
+
+> **Note:** Currently in GraphQL `state` is a separate argument, there is no
```

Comment:
Currently GraphQL can do both, this was intentional.. I can't recall the exact reason, but I think it was so you could do blanket exclusions and/or inclusions of states while also picking specific named tasks (?)